### ISSUE TITLE
Support Nomad's generic volume/volume_mount syntax

### ISSUE THREAD

**Issue by arianvp** [JanMa/nomad-driver-nspawn#7]
Support something like this:

```
  group "example" {
    count = 1
    volume "nix-store" {
      type      = "host"
      source    = "nix-store"
      read_only = true
    }
    task "example" {
      driver = "nspawn"
      volume_mount {
        volume      = "nix-store"
        destination = "/nix/store"
      }
     config {
      image = "debian"
     }
  }
```

It would be implemented the same as the current `bind` and `bind_read_only` syntax. It sounds easy to add.
Happy to send a PR for this

https://www.nomadproject.io/docs/job-specification/volume
https://www.nomadproject.io/docs/job-specification/volume_mount

I guess we can only implement the `host` driver; as I dont know if CSI ingetrates with nspawn

**Comment by JanMa** [comment]
Hey,

adding support to mounting volumes via the proper config options is definitely a
good idea :+1: . I think it might already work for `host` volumes when you make use
of the `bind` or `bind_read_only` stanzas. But then you'd have to know the
actual path of the host volume on the host which is of course not ideal.

I am also not sure if it would be possible to integrate the CSI plugins. I need
to take a closer look at how they work first.

**Comment by JanMa** [comment]
Issue was solved in #13
